### PR TITLE
Python 3.10 support 🥳 

### LINKED ISSUE #243: Py3.10 support
Tried this `virtualenv` today and failed on `llvmlite` install.

Turns out the summary of what we're waiting on is probably the exact same as this minus `pytorch` 😂 

https://github.com/PrincetonUniversity/PsyNeuLink/issues/2164

Particularly we need this release asap..
https://github.com/numba/llvmlite/issues/792

### PR DESCRIPTION
**Pull request by goodboy**
Closes #243.

Starting point for being able to do all the awesome stuff listed in our wish list issues:
- https://github.com/goodboy/tractor/issues/207
- support for the new [structured pattern matching syntax (pep 636)](https://www.python.org/dev/peps/pep-0636/)
  - [`msgspec` already natively supports it](https://github.com/jcrist/msgspec/issues/28)
  - out `Qt` input [processing loops](https://github.com/pikers/piker/blob/master/piker/ui/_interaction.py#L103) would be greatly simplified 🏄🏼 
  
 A follow up feature issue (#272) should be made for the `match:` stuff and a list of specific spots in the code base pointed out to get started 🚀

### CODE REVIEW

**Comment by goodboy** [comment]
Still seeing this from non-crash causing callbacks:

```python
Traceback (most recent call last):
  File "/home/goodboy/repos/piker/piker/ui/_forms.py", line 108, in sizeHint
    psh.setWidth(chars_w_pxs * scale)
TypeError: setWidth(self, int): argument 1 has unexpected type 'float'
Traceback (most recent call last):
  File "/home/goodboy/repos/piker/piker/ui/_forms.py", line 108, in sizeHint
    psh.setWidth(chars_w_pxs * scale)
TypeError: setWidth(self, int): argument 1 has unexpected type 'float'
```

**Review by guilledk — APPROVED** [review]

**Comment by goodboy** [comment]
@guilledk added `msgspec` as dep for now since it's a substantial perf boost and likely what most `tractor` peeps are using.

File: piker/ui/_l1.py [hunk 1]
```diff
@@ -199,7 +199,14 @@ def draw(
         elif self._orient_v == 'top':
             lp, rp = rect.bottomLeft(), rect.bottomRight()
 
-        p.drawLine(lp.x(), lp.y(), rp.x(), rp.y())
+        p.drawLine(
```
Comment (goodboy) [inline piker/ui/_l1.py:202]: Wonder if this is fastest?  🤷🏼

**Comment by goodboy** [comment]
Lol merged to wrong branch...

**Pull request merged**